### ISSUE TITLE
ARIA improvements for +more events link and popover

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#6523]
### The +more link itself

 - **it should be tabbable:** currently it is NOT because it is an [`<a>` tag without an href](https://www.deque.com/blog/accessible-aria-buttons/). One solution is to add `href='#'` or `href="javascript:void(0)"` but a cleaner solution IMO is to only add `tabindex='0'`, then detect a click **as well as** a keypress of ENTER or SPACE. In future versions of FullCalendar we will use a link-like `<button>` element instead.
 - **aria-expanded** when the popover is open (see [disclosures](https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure))
 - **aria-controls** that points to the popover div
 - **aria-label**, determined by a new option. Default defined in the locale:

```js
{
  moreLinkText: "more", // an existing property, results in "+3 more", also accepts function
  moreLinkAriaLabel(n) { // the new property
    return "Show " + n + " more event" + (n === 1 ? "" : "s")
  }
}
```

### The popover

- `aria-labeledby` on the root popover div pointing to the **title** within the popover (with text like "August 31st, 2021")
- the X icon
  - should be tabbable. add `tabindex='0'` and detect mouse click, pressing SPACE, and pressing ENTER, similar to the +more link
  - a keypress of the ESC key should close the popover
  - `aria-label` determined by a new options. Default defined in the locale:

```js
{
  closeAriaLabel: 'Close'
}
```

Get inspired by the [w3c modal dialog example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html)

**Comment by arshaw** [comment]
The mentioned improvements were implemented and released in [v5.10.0](https://github.com/fullcalendar/fullcalendar/releases/tag/v5.10.0)

Instead of `moreLinkAriaLabel` I chose `moreLinkHint`.

Instead of `closeAriaLabel` I chose `closeHint`.
